### PR TITLE
mesa: 20.2.4 -> 20.3.1

### PR DESCRIPTION
**Pull request by primeos**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -58,38 +58,7 @@ stdenv.mkDerivation {
     ./missing-includes.patch # dev_t needs sys/stat.h, time_t needs time.h, etc.-- fixes build w/musl
     ./opencl-install-dir.patch
     ./disk_cache-include-dri-driver-path-in-cache-key.patch
-  ]
-    ++ lib.optionals stdenv.hostPlatform.isMusl [
-      # Fix `-Werror=int-conversion` pthread warnings on musl.
-      # TODO: Remove when https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/6121 is merged and available
-      (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mesa/default.nix:65]: Please remove fetchpatch from inputs.

**Comment by alapshin** [comment]
Is there are any plans of migrating to LLVM 11 to enable support for new AMD cards
https://www.phoronix.com/scan.php?page=news_item&px=AMDGPU-LLVM-11-Navi-2-GFX1030 ?

**Comment by primeos** [comment]
@alapshin I wasn't aware of that, thanks! IIRC upstream CI mainly uses LLVM 9 but Arch already uses LLVM 11 so I think it should be fine to switch to `llvmPackages_latest` to avoid such issues in the future.

**Comment by vcunat** [comment]
Yes, I think mesa generally supports the currently newest LLVM and sometimes it implies additional features in drivers.  This PR seems a good moment when to switch.

**Comment by jansol** [comment]
[20.3.1 is out](https://docs.mesa3d.org/relnotes/20.3.1.html)

**Comment by primeos** [comment]
Tested `r600` and `iris`.

**Pull request merged**